### ISSUE TITLE
Triple slash type reference doesn't use baseUrl/typeRoots

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#35365]
In a directory named `test`:

```ts
// @Filename: tsconfig.json
{
    "compilerOptions": {
        "baseUrl": "../",
        "typeRoots": [ "../" ],
        "types": [],
        "moduleResolution": "node",
        "module": "commonjs"
    },
    "files": [
        "welove.ts",
    ]
}
// @Filename: ex.d.ts
interface F {
  (): void
}
// @Filename: welove.ts
/// <reference types="test/ex" />
declare var f: F
f()
```

Expected: No errors; reference to `F` is fine in `welove.ts`.
Actual: "Cannot find type definition file for 'test/ex'.

Related things that work: `"../test/ex", "./ex"`.
Also, `import { A } from "test/foo"` works -- imports use baseUrl/typeRoots correctly.

**Comment by RyanCavanaugh** [comment]
Fixed in TypeScript 5.1.6. The original project reports `TS2688: Cannot find type definition file for 'test/ex'` in 5.0.4, while 5.1.6 resolves `test/ex`, brings `F` into scope, and compiles without errors.

Current native TypeScript requires removing the retired `baseUrl` and `moduleResolution: "node"` options from this configuration; with those migration-only changes, the same `/// <reference types="test/ex" />` resolves without errors.
